zerver/lib/mention.py

```diff
@@ -139,24 +139,27 @@ def user_mention_matches_wildcard(mention: str) -> bool:
     return mention in wildcards
 
 
-def extract_mention_text(m: Match[str]) -> Tuple[Optional[str], bool]:
+def extract_mention_text(m: Match[str]) -> Tuple[Optional[str], bool, bool]:
     text = m.group("match")
     if text in wildcards:
-        return None, True
-    return text, False
+        return None, True, (text in topic_wildcards)
```

Comment:
I'm not sure we want `@topic` to have the `has_wildcard` flag set; I think we should have `None, False, True` for a topic wildcard. This achieves two things:
* `@topic` won't do anything until we actually have implemented it further, I think (I believe your implementation might make it work like `@all`?)
* I think data plumbing will be simpler if we don't mingle this with another field.

Concretely, I think this means just not adding it to `wildcards` at the top of this file.